org.eclipse.scout.rt.jackson.test/src/test/java/org/eclipse/scout/rt/jackson/dataobject/JsonDataObjectsSerializationTest.java

```diff
@@ -2919,10 +2946,297 @@ public void testSerializeWithUnknownNestedEntity6() throws Exception {
+  @Test
+  public void testDeserializeType_IDoEntity1() throws Exception {
+    String json = "{\"foo\" : \"bar\"}";
+    IDoEntity marshalled = s_dataObjectMapper.readValue(json, IDoEntity.class);
+    assertEquals(DoEntity.class, marshalled.getClass());
+  }
+
+  /**
+   * JSON with type information (class available), expect generic IDoEntity.
+   */
+  @Test
+  public void testDeserializeType_IDoEntity2() throws Exception {
+    String json = "{\"_type\" : \"TestItem\"}";
+    IDoEntity marshalled = s_dataObjectMapper.readValue(json, IDoEntity.class);
+    assertEquals(TestItemDo.class, marshalled.getClass());
+  }
+
+  /**
+   * JSON with type information (class not known), expect generic IDoEntity.
+   */
+  @Test
+  public void testDeserializeType_IDoEntity3() throws Exception {
+    String json = "{\"_type\" : \"UnknownEntity\"}";
+    IDoEntity marshalled = s_dataObjectMapper.readValue(json, IDoEntity.class);
+    assertEquals(DoEntity.class, marshalled.getClass());
+  }
+
+  /**
+   * JSON with type information (class available), expect nested generic IDoEntity.
+   */
+  @Test
+  public void testDeserializeType_IDoEntity4() throws Exception {
+    String json = "{\"_type\" : \"TestNestedRaw\", \"iDoEntity\" : {\"_type\" : \"TestItem\"}}";
+    TestNestedRawDo marshalled = s_dataObjectMapper.readValue(json, TestNestedRawDo.class);
+    assertEquals(TestItemDo.class, marshalled.getIDoEntity().getClass());
+  }
+
+  /**
+   * JSON with type information (class not known), expect nested generic IDoEntity.
+   */
+  @Test
+  public void testDeserializeType_IDoEntity5() throws Exception {
+    String json = "{\"_type\" : \"TestNestedRaw\", \"iDoEntity\" : {\"_type\" : \"UnknownEntity\"}}";
+    TestNestedRawDo marshalled = s_dataObjectMapper.readValue(json, TestNestedRawDo.class);
+    assertEquals(DoEntity.class, marshalled.getIDoEntity().getClass());
+  }
+
+  /**
+   * JSON without type information, expect generic IDataObject.
+   */
+  @Test
+  public void testDeserializeType_IDataObject1() throws Exception {
+    String json = "{\"foo\" : \"bar\"}";
+    IDataObject marshalled = s_dataObjectMapper.readValue(json, IDataObject.class);
+    assertEquals(DoEntity.class, marshalled.getClass());
+  }
+
+  /**
+   * JSON with type information (class available), expect generic IDataObject.
+   */
+  @Test
+  public void testDeserializeType_IDataObject2() throws Exception {
+    String json = "{\"_type\" : \"TestItem\"}";
+    IDataObject marshalled = s_dataObjectMapper.readValue(json, IDataObject.class);
+    assertEquals(TestItemDo.class, marshalled.getClass());
+  }
+
+  /**
+   * JSON with type information (class not known), expect generic IDataObject.
+   */
+  @Test
+  public void testDeserializeType_IDataObject3() throws Exception {
+    String json = "{\"_type\" : \"UnknownEntity\"}";
+    IDataObject marshalled = s_dataObjectMapper.readValue(json, IDataObject.class);
+    assertEquals(DoEntity.class, marshalled.getClass());
+  }
+
+  /**
```

Comment:
Idea: improve tests so that for untyped the "_type" attribute is checked if it's equal to the expected and for typed, that it doesn't exists? Not sure though if that is checked somewhere else already.

Reply:
Added assertions